### PR TITLE
114 update rails

### LINKED ISSUE #114: Update Rails to 4.2.7.1
Github is telling us that there are vulnerabilities in our current version, and we should update to at least 4.2.7.1

### PR DESCRIPTION
**Pull request by rebeccafae**
## Description

Update to Rails 4.2.7 because of known security vulnerabilities in our old version

## Github Issue
Fixes #114

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 module ApplicationHelper
   def default_omniauth_authorize_path
-    user_omniauth_authorize_path(ENV['DEFAULT_OMNIAUTH_PROVIDER'] || :linkedin)
+    if Rails.env.production?
+      user_linkedin_omniauth_authorize_path
```
Comment (rebeccafae) [inline app/helpers/application_helper.rb:4]: optionally, `send("user_#{ENV['EFAULT_OMNIAUTH_PROVIDER'] || 'linkedin')}_omniauth_authorize_path")`

File: app/models/mentorship.rb [hunk 2]
```diff
@@ -41,8 +41,8 @@ def choose_mentor
   end
 
   def send_mail
-    UserMailer.mentor_mail(self).deliver
-    UserMailer.mentee_mail(self).deliver
+    UserMailer.mentor_mail(self).deliver_now
```
Comment (rebeccafae) [inline app/models/mentorship.rb:46]: not later?

File: config/initializers/active_admin.rb [hunk 3]
```diff
@@ -142,7 +142,7 @@
   # You can add before, after and around filters to all of your
   # Active Admin resources and pages from here.
   #
-  # config.before_filter :do_something_awesome
+  # config.before_action :do_something_awesome
```
Comment (rebeccafae) [inline config/initializers/active_admin.rb:145]: this is a comment

File: spec/controllers/mentorships_controller_spec.rb [hunk 4]
```diff
@@ -29,7 +29,7 @@
     end
 
     it 'does not destroy other users mentorships' do
-      expect(Mentorship.find(other_pending_mentorship)).not_to be_nil
+      expect(Mentorship.find(other_pending_mentorship.id)).not_to be_nil
```
Comment (rebeccafae) [inline spec/controllers/mentorships_controller_spec.rb]: expect the find to `not throw`, or else use find_by_id

**Review by rebeccafae — COMMENTED** [review]
~ @mrgenixus

**Review by mrgenixus — APPROVED** [review]

**Pull request merged**